src/server/handlers/consents/{ID}.ts

```diff
@@ -0,0 +1,93 @@
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Ahan Gupta <ahangupta@google.com>
+
+ --------------
+ ******/
+import { Request, ResponseToolkit, ResponseObject } from '@hapi/hapi'
+import { Consent } from '../../../model/consent'
+import { Logger } from '@mojaloop/central-services-logger'
+import { retrieveValidConsent, updateConsentCredential, putConsents, ConsentCredential, checkCredentialStatus } from '../../../domain/consents/consents'
+import { IncorrectChallengeError, IncorrectStatusError } from '../../../domain/errors'
+import { verifySignature } from '../../../lib/challenge'
+import { NotFoundError } from '../../../model/errors'
+import { Enum } from '@mojaloop/central-services-shared'
+
+export async function retrieveUpdateAndPutConsent (id: string, challenge: string,
+  credentialStatus: string, signature: string,
+  publicKey: string, requestCredentialId: string,
+  request: Request): Promise<void> {
+  let consent: Consent
+  try {
+    consent = await retrieveValidConsent(id, challenge)
+    /* Checks if incoming credential status is of the correct form */
+    await checkCredentialStatus(credentialStatus, id)
+    try {
+      if (!verifySignature(challenge, signature, publicKey)) {
+        Logger.push({ consentId: id })
+        Logger.error('Invalid Challenge')
+        /* TODO, make outbound call to PUT consents/{ID}/error to be addressed in ticket number 355 */
+        return
+      }
+      const credential: ConsentCredential = {
+        credentialId: requestCredentialId,
+        credentialStatus: 'ACTIVE',
+        credentialPayload: publicKey
+      }
+      await updateConsentCredential(consent, credential)
+
+      /* Outbound PUT consents/{ID} call */
+      putConsents(consent, signature, publicKey, request)
+    } catch (error) {
+      Logger.push(error)
+      Logger.error('Error: Outgoing call with challenge credential NOT made to PUT consents/' + id)
+      /* TODO, make outbound call to PUT consents/{ID}/error to be addressed in ticket number 355 */
+      return
+    }
+  } catch (error) {
+    if (error instanceof IncorrectChallengeError || error instanceof IncorrectStatusError || error instanceof NotFoundError) {
+      Logger.push(error)
+    }
+    Logger.push(error)
+    Logger.error('Error in retrieving consent.')
+    /* TODO, make outbound call to PUT consents/{ID}/error to be addressed in ticket number 355 */
+  }
+}
+
+export default async function put (request: Request, h: ResponseToolkit): Promise<ResponseObject> {
+  const id = request.params.id
+  // @ts-ignore
+  const requestPayloadCredential = request.payload.credential
+  const [signature, publicKey, challenge, requestCredentialId, credentialStatus] = [
```

Comment:
might be wrong here but thought it with `{}` not `[]`

Reply:
@akapur99 it is an array decomposing

Reply:
but still, it looks weird